packages/story-editor/src/components/library/panes/media/common/insertionMenu.js

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+import {
+  useRef,
+  useMemo,
+  useCallback,
+  useState,
+} from '@googleforcreators/react';
+import { __ } from '@googleforcreators/i18n';
+import { v4 as uuidv4 } from 'uuid';
+import {
+  Icons,
+  Menu,
+  PLACEMENT,
+  Popup,
+  useKeyDownEffect,
+} from '@googleforcreators/design-system';
+import {
+  getSmallestUrlForWidth,
+  ResourcePropTypes,
+} from '@googleforcreators/media';
+
+/**
+ * Internal dependencies
+ */
+import getElementProperties from '../../../../canvas/utils/getElementProperties';
+import useStory from '../../../../../app/story/useStory';
+import { ActionButton } from '../../shared';
+import useRovingTabIndex from '../../../../../utils/useRovingTabIndex';
+import useFocusCanvas from '../../../../canvas/useFocusCanvas';
+
+const DropDownContainer = styled.div`
+  margin-top: 10px;
+  min-width: 160px;
+`;
+
+const MenuContainer = styled.div`
+  z-index: 1;
+`;
+
+const menuStylesOverride = css`
+  min-width: 160px;
+  margin-top: 0;
+  li {
+    display: block;
+  }
+`;
+
+// This is used for nested roving tab index to detect parent siblings.
+const BUTTON_NESTING_DEPTH = 3;
+const MENU_OPTIONS = {
+  INSERT: 'insert',
+  ADD_BACKGROUND: 'addBackground',
+};
+
+/**
+ * Get a More icon that displays a dropdown menu on click.
+ *
+ * @param {Object} props Component props.
+ * @param {Object} props.resource Selected media element's resource object.
+ * @param {boolean} props.display Whether the more icon should be displayed.
+ * @param {Function} props.onInsert Callback for inserting media.
+ * @param {number} props.width Media width.
+ * @param {number} props.index Element index in the gallery.
+ * @param {boolean} props.isLocal If the menu is for local or 3p media.
+ * @return {null|*} Element or null if should not display the More icon.
+ */
+function InsertionMenu({ resource, display, onInsert, width, index, isLocal }) {
+  const insertButtonRef = useRef();
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const onMenuOpen = useCallback((e) => {
+    e.stopPropagation();
+    setIsMenuOpen(true);
+  }, []);
+  const onMenuCancelled = useCallback(() => setIsMenuOpen(false), []);
```

Comment:
When you press esc to close the insertion menu, focus is moved to the canvas rather than back to the media item.

How to reproduce:
* Use tab to focus media item
* Press <kbd>enter</kbd> to open insertion menu
* Press <kbd>esc</kbd> to close insertion menu
* Observe that focus is now on canvas, not media item

The solution here is potentially just to force focus to the media item in this `onMenuCancelled` callback?

Reply:
There's an issue to follow-up in this: #10607 
(This was an issue before this PR already, too)